content/doc/book/system-administration/FIPS-140.adoc

```diff
@@ -0,0 +1,54 @@
+How to configure the servlet container, JVM and host are out of scope of the Jenkins community project as this is a complex area with many pitfalls and gotchas.
+Some Jenkins features may not work or be disabled.
+
+== Plugins
+
+Plugins may or may not honour a request to run in FIPS-140 compliance mode.
+As such before installing or upgrading any plugin you should make checks of the plugins code to ensure it adheres to the FIPS-140 standard.
+
+== What FIPS-140 mode does
+
+Enabling <<managing/system-properties#jenkins-security-FIPS140-COMPLIANCE, FIPS mode>> provides a hint to Jenkins and any plugins that have opted in that they should prefer cryptographic algorithms that *may*.footnote:[Algorithms are not approved, rather an implementation of an algorithm is.  
+However the implementation used at runtime will depend on the JVM, JVM configuration, as well as the host OS.
+As this is outside the scope of the Jenkins project the algorithms targeted are known to be available from at least one link:https://csrc.nist.gov/projects/cryptographic-module-validation-program/validated-modules/search[FIPS compliant provider], namely the link:https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/3514[BouncyCastle FIPS library].] be FIPS approved.
+This may mean that some features are disabled entirely, or even may use a less secure (but compliant) form of cryptography than normal.
+
+== What FIPS-140 mode does not do
+
+FIPS mode is not a magic bullet, if any code from the JVM, servlet container, Jenkins or any plugin asks for a non compliant algorithm this will still be the case, and the request may well be honoured.
+For example this mode does not and can not configure the JVM, so TLS connections to external secure web sites may still use non compliant cryptography.
+Further, Jenkins cannot ensure that plugins even use encryption at all when appropriate.
+At the end of the day, just because Jenkins and plugins run when FIPS mode is enabled does not mean that it adheres to the USA government standard.
+
+== How to run a fully FIPS-140 compliant Jenkins
+
+As mentioned above the host, JVM and servlet container all need to be configured appropriately to ensure Jenkins is FIPS-140 compliant.  
+Additionally extreme care should be taken when installing or upgrading plugins as they may or may not be FIPS-140 compliant, and they may introduce code that is non compliant or otherwise change the JVM configuration in such a way that breaks compliance.
+
+The Jenkins community does not support Jenkins FIPS mode, and due to the complex nature of JVM and servlet configuration that can change between versions, will not provide documentation for this.
+Should you wish to run Jenkins in a FIPS compliant way it is therefore recommended to obtain support from a commercial vendor that will support you in doing so.
```

Comment:
```suggestion
If you want to run Jenkins in a FIPS-compliant environment, it is recommended that you obtain support from a commercial vendor.
```

Reply:
it is not the environment that you want to run Jenkins in - it is that you want to run Jenkins FIPS-140  Compliant.

```suggestion
If you need to run Jenkins in a way that it is FIPS compliant, it is recommended that you obtain support from a commercial vendor.
```